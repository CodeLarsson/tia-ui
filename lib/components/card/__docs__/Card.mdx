import { Canvas, Meta } from '@storybook/blocks';
import Example from './Example.tsx';
import * as Card from './Card.stories.tsx';

<Meta of={Card} title="Card" />

# Card

Card component with different props.

#### Example

<Canvas of={Card.Elevated} />

## Usage

```ts
import {Card} from "tia-ui";

const Example = () => {
  return (
      <Card
        type="elevated",
        title="Elevated Card"
        icon={() => <Icon name="User" />}
      />
  );
};

export default Example;
```

#### Arguments

- **type** - Type of card, valid values are 'elevated' | 'outlined' | 'filled', default value is 'elevated'
