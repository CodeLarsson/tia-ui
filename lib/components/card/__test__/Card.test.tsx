import React from 'react';
import { describe, expect, it } from 'vitest';
import { render, screen } from '@testing-library/react';
import Card from '../Card';

describe('Card component', () => {
  it('Card should render correctly', () => {
    // Arrange
    render(<Card />);
    const card = screen.getByRole('region');

    // Assert
    expect(card).toBeInTheDocument();
  });

  it('Card should render as elevated by default', () => {
    // Arrange
    render(<Card />);
    const card = screen.getByRole('region');

    // Assert
    expect(card.classList).toContain('tia-shadow-xl');
  });

  it('Card should match snapshot', () => {
    // Arrange
    const { container } = render(<Card />);

    // Aassert
    expect(container).toMatchSnapshot();
  });
});
