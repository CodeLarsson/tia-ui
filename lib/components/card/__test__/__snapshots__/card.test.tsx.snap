// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card component > Card should match snapshot 1`] = `
<div>
  <div
    class="tia-p-6 tia-max-w-sm tia-rounded-xl tia-space-y-2 tia-text-gray-800" tia-bg-primary tia-shadow-lg"
    role="region"
  >
    <div
      class="tia-flex tia-items-center tia-space-x-2"
    />
    <div />
  </div>
</div>
`;
